### PR TITLE
Fix python 3.11 test issues

### PR DESCRIPTION
**Pull request by drroe**
Try to fix some test errors that seem to show up with python 3.11. @hainm please especially check the change in `c_trajectory.pyx1. I'm not completely certain of the intent behind the original code, but changing it to something like `if not idxs` did not fix the error.

### CODE REVIEW

File: pytraj/trajectory/c_traj/c_trajectory.pyx [hunk 1]
```diff
@@ -318,9 +318,10 @@ cdef class TrajectoryCpptraj:
             return self.tmpfarray
         else:
             # not is a slice or string or AtomMask
-            if idxs == ():
-                return self
-            elif isinstance(idxs, tuple):
+            #if idxs == ():
+            #    return self
+            #el
```
Comment (hainm) [inline pytraj/trajectory/c_traj/c_trajectory.pyx:323]: @drroe This code is trying to mimic the numpy array slicing

```python
In [3]: import numpy as np                                                                                                                                            

In [4]: arr = np.array([1, 2, 3])                                                                                                                                     

In [5]: arr[()]                                                                                                                                                       
Out[5]: array([1, 2, 3])
```

What's the failure message before your change?

Comment (drroe) [reply pytraj/trajectory/c_traj/c_trajectory.pyx:323]:
```python
______________________________ TestSlicingTrajectory.test_array_like _______________________________

self = <test_trajectory.test_fancy_indexing.TestSlicingTrajectory object at 0x7fbfd0851a90>

    def test_array_like(self):
        traj = pt.iterload(fn('Tc5b.x'), fn('Tc5b.top'))
        xyz_source = traj.xyz[:].copy()
        traj_mem = traj[:]
    
        # slicing with list or array
        indices = [1, 2, 3]
        fa = traj[indices]
        fa2 = traj_mem[indices]
        fa3 = traj[range(1, 4)]
        fa4 = traj_mem[range(1, 4)]
        assert isinstance(fa, Trajectory)
        # from TrajectoryIterator
        aa_eq(fa[0].xyz, traj[1].xyz)
        aa_eq(fa[1].xyz, traj[2].xyz)
        # from Trajectory
        aa_eq(fa2[1].xyz, traj[2].xyz)
        aa_eq(fa2[0].xyz, traj[1].xyz)
    
        # a list with one element
        assert isinstance(traj[[1,]], Trajectory)
        aa_eq(traj[[1,]].xyz, traj[1:2].xyz)
    
        # from "range"
        aa_eq(fa3[1].xyz, traj[2].xyz)
        aa_eq(fa3[0].xyz, traj[1].xyz)
        aa_eq(fa4[1].xyz, traj[2].xyz)
        aa_eq(fa4[0].xyz, traj[1].xyz)
    
        # tuple
        aa_eq(traj[(1,)].xyz, xyz_source[1])
    
        # boolean
        bool_arr = np.array([False]*traj.n_frames)
        bool_arr[0] = True
        bool_arr[2] = True
>       new_xyz = traj[bool_arr].xyz

test_trajectory/test_fancy_indexing.py:49: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

>   ???
E   ValueError: operands could not be broadcast together with shapes (10,) (0,)

pytraj/trajectory/c_traj/c_trajectory.pyx:321: ValueError
```
I think the issue is that `idxs` ends up being a 1D array, but `()` ends up resolving to 0D, which I guess is cool in some versions of numpy/python and not cool in others.

Comment (drroe) [reply pytraj/trajectory/c_traj/c_trajectory.pyx:323]: FYI here are the errors in the wild: https://github.com/Amber-MD/cpptraj/actions/runs/5646398379/job/15294214272?pr=1042

**Review by hainm — APPROVED** [review]
@drroe Please go ahead merging once you're ready. Thanks.

**Pull request merged**